pkg/broker/handler/pool/pool_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pool
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestSyncPool(t *testing.T) {
+	wantErr := fmt.Errorf("error returned from fakeSyncPool")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	syncPool := &FakeSyncPool{}
+
+	t.Run("StartSyncPool returns error", func(t *testing.T) {
+		// Add some brokers with their targets.
+		tctx := context.WithValue(ctx, "error", "true")
```

Comment:
nit: it's better if your fake has field to indicate whether you want an error or not.

```
type fakePool struct {
	returnErr bool
}
```